### PR TITLE
Change log source from docker to journalctl

### PR DESCRIPTION
**Pull request by codewithcheese**
- `logMonitor.start()` call in supervisor `init`, spawns `journalctl` process to follow all container logs
- `logMonitor.attach()` backfills logs since `lastSentTimestamp` with a new `journalctl` process using `since` argument
- `logMonitor.detach()` is called on docker `destroy` event

Change-type: minor
Signed-off-by: Thomas Manning <thomasm@balena.io>

### CODE REVIEW

File: src/logging/monitor.ts [hunk 1]
```diff
@@ -1,24 +1,90 @@
...
+
 /**
  * This class provides a wrapper around the database for
  * saving the last timestamp of a container
  */
-export class LogMonitor {
+class LogMonitor {
+
+	private hooks: { [containerId: string]: MonitorHook } = {};
+	private journalctl: ChildProcess;
```
Comment (CameronDiver) [inline src/logging/monitor.ts:45]:
```suggestion
	private journalctl: ReturnType<typeof journald.spawnJournalCtl>;
```

Comment (codewithcheese) [reply src/logging/monitor.ts]: updated

**Review by CameronDiver — APPROVED** [review]

**Comment by codewithcheese** [comment]
@balena-ci I self-certify!
